compiler/sem/varpartitions.nim

```diff
@@ -807,6 +807,10 @@ proc computeLiveRanges(c: var Partitions; n: PNode) =
       else:
         for i in 0..<child.len-2:
           registerVariable(c, child[i])
+          if child.kind == nkVarTuple:
+            # an unpacking definition where the rhs is not a literal tuple
+            # construction. The vars are always owning
```

Comment:
From the PR description, it seems to me that we need this because `mirgen` assumes ownership. I think it's worth adding a note about that, since that would open the door to undoing this in the future (assuming that it's not gonna be done via MIR instead).

Reply:
I actually considered mentioning `mirgen` when writing the comment, and in the past, I employed such mentions/links quite often, but I no longer think that this a good idea, because:
* mentioning expectations of other not-directly-related processing to justify/explain some local behaviour is a workaround to no well-defined IR existing. Continued use of such mentions only exacerbates the ad-hoc semantics of AST
* these manual links are very susceptible to becoming stale, and a lot of them already did